### PR TITLE
Convert `InfoBond::ArpValidate` into an enum.

### LINKED ISSUE #80: Convert `InfoBond::ArpValidate(u32)` to enum
```rust
const BOND_ARP_VALIDATE_NONE: u32 = 0;
const BOND_ARP_VALIDATE_ACTIVE: u32 = 1 << BOND_STATE_ACTIVE as u32;
const BOND_ARP_VALIDATE_BACKUP: u32 = 1 << BOND_STATE_BACKUP as u32;
const BOND_ARP_VALIDATE_ALL: u32 =
    BOND_ARP_VALIDATE_ACTIVE | BOND_ARP_VALIDATE_BACKUP;
const BOND_ARP_FILTER: u32 = BOND_ARP_VALIDATE_ALL + 1;
const BOND_ARP_FILTER_ACTIVE: u32 = BOND_ARP_VALIDATE_ACTIVE | BOND_ARP_FILTER;
const BOND_ARP_FILTER_BACKUP: u32 = BOND_ARP_VALIDATE_BACKUP | BOND_ARP_FILTER;
```

### PR DESCRIPTION
**Pull request by Gilnaa**
Closes #80

### CODE REVIEW

File: src/link/link_info/bond.rs [hunk 1]
```diff
@@ -199,6 +211,65 @@ impl std::fmt::Display for BondMode {
...
+    #[default]
+    None,
+    Active,
+    Backup,
+    All,
+    Filter,
+    FilterActive,
+    FilterBackup,
+    Other(u32),
+}
```
Comment (Gilnaa) [inline src/link/link_info/bond.rs:225]: @cathay4t 
In general, does it make sense to add the kernel documentation to these enums?

e.g. based on https://www.kernel.org/doc/html/v6.4-rc7/networking/bonding.html
```suggestion
/// Specifies whether or not ARP probes and replies should be validated in any mode that supports arp monitoring,
/// or whether non-ARP traffic should be filtered (disregarded) for link monitoring purposes.
#[derive(Debug, Clone, Copy, Eq, PartialEq, Default)]
pub enum ArpValidate {
    /// No validation or filtering is performed.
    #[default]
    None,
    /// Validation is performed only for the active slave.
    Active,
    /// Validation is performed only for backup slaves.
    Backup,
    /// Validation is performed for all slaves.
    All,
    /// Filtering is applied to all slaves. No validation is performed.
    Filter,
    /// Filtering is applied to all slaves, validation is performed only for the active slave
    FilterActive,
    /// Filtering is applied to all slaves, validation is performed only for backup slaves.
    FilterBackup,
    Other(u32),
}
```

**Comment by Gilnaa** [comment]
Also, it seems that these PRs steadily lower the code-coverage.
What meaningful tests can I add?
Run an `ip link add bond0 type bond mode blablablah` and make sure we produce an equivalent output?

**Comment by cathay4t** [comment]
> Also, it seems that these PRs steadily lower the code-coverage. What meaningful tests can I add? Use `nlmon` to record `ip link add bond0 type bond mode blablablah` and make sure we produce an equivalent output?

I have document on how to capture netlink package and convert it into Vec<u8> at:

https://github.com/rust-netlink/netlink-packet-route/blob/main/README.md

You may setup up a bond with arp_validate value changed to non-default one. Then capture the netlink package kernel replied to the request of `ip -d link show bond0`. The unit test should parse this Vec<u8> into structs/enums, and also `emit()` should generate identical `Vec<u8>` as kernel did.

> 
> Is there place for tests that open an actual socket and try to create rtnl objects; is this problematic with CI running us in a non-privileged context?

For this crate, there is no real netlink communication to kernel. Because this crate only handle parsing and emitting netlink package itself, no socket communication code included. All tests in this crate are unit test, no privilege required.

**Comment by little-dude** [comment]
@Gilnaa I just added a test in my own MR, in case you want to see how it looks: https://github.com/rust-netlink/netlink-packet-route/pull/98

I used wireshark to get the hex dump of a captured netlink packet, and isolate the different packet parts. It's very tedious, and you don't have to go into so much detail though.

**Comment by Gilnaa** [comment]
Thank you both!

Unfortunately life got in the way;
will hopefully get back to it eventually

**Comment by Gilnaa** [comment]
> Unfortunately life got in the way; will hopefully get back to it eventually

Life got out of the way!

I've added a test as instructed; dump is taken of `tcpdump`, with the last few fields removed for brevity, and the length patched in the header.

--- 
I did notice that serializing `stats64` will result in a slightly different payload (I think the length field was different, probably counted padding or something).
Unrelated to this change, though.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: src/link/tests/bond.rs [hunk 2]
```diff
@@ -56,14 +58,14 @@ fn test_bond_link_info() {
...
-                InfoBond::Mode(BondMode::BalanceRr),
+                InfoBond::Mode(BondMode::default()),
                 InfoBond::MiiMon(0),
                 InfoBond::UpDelay(0),
                 InfoBond::DownDelay(0),
                 InfoBond::PeerNotifDelay(0),
                 InfoBond::UseCarrier(1),
                 InfoBond::ArpInterval(0),
-                InfoBond::ArpValidate(0),
+                InfoBond::ArpValidate(ArpValidate::default()),
```
[the remark this answers is not in the document]
Comment (Gilnaa) [reply src/link/tests/bond.rs]: Done

File: src/link/tests/bond.rs [hunk 3]
```diff
@@ -149,3 +151,83 @@ fn test_bond_port_link_info() {
...
+        NetlinkHeader::default(),
+        NetlinkPayload::from(RouteNetlinkMessage::NewLink(expected)),
+    );
+    packet.header.sequence_number = 1716697509;
+    packet.header.port_number = 4040503;
+    packet.finalize();
+    let mut buf = vec![0u8; packet.buffer_len()];
+    packet.serialize(&mut buf);
+
+    assert_eq!(raw, buf);
```
[the remark this answers is not in the document]
Comment (Gilnaa) [reply src/link/tests/bond.rs]: Ohhh great idea.
Done

**Review by cathay4t — APPROVED** [review]

**Comment by cathay4t** [comment]
I have amend the patch:
 * Rebased to latest main branch.
 * Changed `enum ArpValidate` to `enum BondArpValidate`.

**Pull request merged**